pkg/apis/build/v1alpha2/image_builds.go

```diff
@@ -31,9 +32,15 @@ const (
 
 type BuildReason string
 
-func (im *Image) Build(sourceResolver *SourceResolver, builder BuilderResource, latestBuild *Build, reasons, changes string, nextBuildNumber int64) *Build {
+func (im *Image) Build(sourceResolver *SourceResolver, builder BuilderResource, latestBuild *Build, reasons, changes string, nextBuildNumber int64, priorityClass string) *Build {
 	buildNumber := strconv.Itoa(int(nextBuildNumber))
-
+	buildAnnotations := map[string]string{
+		BuildReasonAnnotation:  reasons,
+		BuildChangesAnnotation: changes,
+	}
+	if priorityClass != "" {
+		buildAnnotations[BuildPriorityAnnotation] = priorityClass
```

Comment:
It seems like this field is actionably modifying the behavior of the build. Could it be configured on the build.spec?

Reply:
Obviously, that introduces complexity with the apis.

Reply:
+1 spec